collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_bucket.py

```diff
@@ -121,55 +137,76 @@ def get_bucket(module, blade):
 
 def create_bucket(module, blade):
     """Create bucket"""
-    changed = False
-    try:
-        attr = Bucket()
-        attr.account = Reference(name=module.params['account'])
-        blade.buckets.create_buckets(names=[module.params['name']], account=attr)
-        changed = True
-    except Exception:
-        module.fail_json(msg='Object Store Bucket {0}: Creation failed'.format(module.params['name']))
+    changed = True
+    if not module.check_mode:
+        try:
+            attr = Bucket()
+            attr.account = Reference(name=module.params['account'])
+            blade.buckets.create_buckets(names=[module.params['name']], account=attr)
+            api_version = blade.api_version.list_versions().versions
+            if module.params['versioning'] != 'absent' and VERSIONING_VERSION in api_version:
+                try:
+                    blade.buckets.update_buckets(names=[module.params['name']],
+                                                 bucket=BucketPatch(destroyed=True, versioning=module.params['versioning']))
+                except Exception:
+                    module.fail_json(msg='Object Store Bucket {0} Creationed but versioning state failed'.format(module.params['name']))
```

Comment:
Should be `created` instead of `Creationed`.

Reply:
Fixed